scripts/polyrith_sage.py

```diff
@@ -27,8 +27,15 @@ def create_query(type: str, n_vars: int, eq_list, goal_type):
 gens = {eq_list}
 p = P({goal_type})
 I = ideal(gens)
-coeffs = p.lift(I)
-print(json.dumps([polynomial_to_string(c) for c in coeffs]))
+try:
+    coeffs = p.lift(I)
+    n = int(1)
+except Exception as e:
+    if p not in I.radical():
+        raise e
+    n = naive_power_search(p, I)
+    coeffs = (p^n).lift(I)
```

Comment:
You can save a recomputation with
```suggestion
    p_i = p
    for i in count(2):
        p_i *= p
        if p_i in I:
            coeffs = p_i.lift(I)
            break
```